File: Javascript/1.js
// for (let x = 1; x < 101; x++) {
//     if (x%15 == 0) {
//         console.log("FizzBuzz");
//     }
//     else if (x%5 == 0) {
//         console.log("Buzz");
//     }
//     else if (x%3 == 0) {
//         console.log( "Fizz");
//     }
//     else {
//         console.log(x);
//     }
// }
// // the program goes through the x%15 first, and then goes through the x%5 and x%3

// // start with the first hashtag
// // add 1 to the previous hashtag
// // keep adding until there are 7 hashtags
// let y = "#";
// while (y.length <= 7 ) {
//     console.log(y)
//     y+="#";
// }

// function linearSearch (list, targetvalue) {
//     for(let index = 0; index < list.length; index++) {
//         if (list[index] == targetvalue) {
//             return index;
//         }
//     }
// }
// let list = ["apple", "banana", "cherry", "dragonfruid", "strawberry"];
// let targetvalue = "berry";
// let foundindex = linearSearch(list, targetvalue);
// console.log(foundindex);

// function selectionSort (list) {
//     for (let index = 0; index < list.length; index ++) {
//         let minIndex = index;
//         for (let newIndex = index; newIndex < list.length; newIndex ++) {
//             if (list[newIndex] < list[minIndex]) {
//                 minIndex = newIndex;
//             }
//         }
//         let holder = list[index];
//         list[index] = list[minIndex];
//         list[minIndex] = holder;
//     }
//     return list;
// }
// let list = [2,6,2,4,8,4];
// console.log(list);
// console.log(selectionSort(list));

function insertionSort (list) {
    for (let index = 1; index < list.length; index++) {
        let newIndex = index;
        while ((list[newIndex] < list[newIndex -1]) && (newIndex > 0)) {
            let holder = list[newIndex];
            list[newIndex] = list[newIndex -1];
            list[newIndex -1] = holder;
            newIndex--;
        }
    }
    return list;
}
let list = [4,3,5,1,2];
console.log(list);
console.log(insertionSort(list));

File: Javascript/1.py
for x in range (1, 101): #loop through 1 to 100
    if x%15 == 0:
        print("FizzBuzz") #the output will be FizzBuzz when the number devisable by 15 
    elif x%5 == 0:
        print("Buzz") #the output will be Buzz when the number devisable by 5
    elif x%3 == 0:
        print("Fizz") #the output will be Fizz when the number devisable by 3
    else:
        print(x) #the output will be the number when it isn't devisable by 3, 5 and 15
    

def linearSearch(fruit, targetvalue): #to check if the target value matches the fruit in the fruit list
    for y in fruit:
        if y == targetvalue:
            return fruit.index(y)
fruit = ["apple", "banana", "cherry", "dragonfruit", "elderberry"] #list of the fruit
targetvalue = "cherry"
foundFruit = linearSearch(fruit, targetvalue)
print(foundFruit) 

